rust/blockstore/src/types/reader.rs

```diff
@@ -0,0 +1,143 @@
+use crate::arrow::blockfile::ArrowBlockfileReader;
```

Comment:
In the future it would be helpful if reorgs like this were there own PRs so that reviewers can be aware of  "move to another file" vs "this is a change"